server/core/service.py

```diff
@@ -0,0 +1,11 @@
+from abc import abstractmethod
+
+
+class Service(object):
```

Comment:
Yea I'm not quite sure how I want to do services. Another option is with a decorator, but ideally we'd refactor it so that services are created automatically, and all the initialize code is created automatically. I was also thinking maybe `update_data` should be a function on the base class since most of the services seem to use it.

Reply:
Well, if you have it that way, just write what do you define now, and how it should be used, it's interface what you're doing :-)

Reply:
Maybe also cleanup / shutdown, depending on how many services have one. (I only know of the ladder service.)